### DISCUSSION TITLE
How to pass commands to docker image?

### DISCUSSION THREAD

**Discussion by hailwood** [aws/copilot-cli#5714] [Q&A]
When running typesense via docker you use

```
docker run -p 8108:8108 \
            -v$(pwd)/typesense-data:/data typesense/typesense:0.25.2 \
            --data-dir /data \
            --api-key=$TYPESENSE_API_KEY \
            --enable-cors
```

When specifying my `image.location` as `typesense/typesense:0.25.2` how can I pass through the `--enable-cors` flag etc?

**Comment by iamhopaul123** [comment]
Hello @hailwood. Sorry I am not very familiar with what these flags do

```
            --data-dir /data \
            --api-key=$TYPESENSE_API_KEY \
            --enable-cors
```

Are they valid flags for `docker run` 🤔

**Comment by hailwood** [comment]
Hi @iamhopaul123 

They're flags for the typesense-server program.

So the container has an ENTRYPOINT of `/opt/typesense-server` so what ends up getting run inside the container with that command is
```
/opt/typesense-server --data-dir /data --api-key=$TYPESENSE_API_KEY --enable-cors
```

It looks like there might be plans for it or something based on https://aws.github.io/copilot-cli/docs/include/command/

But I tried doing

```
image:
  command: /opt/typesense-server --data-dir /data --api-key=$TYPESENSE_API_KEY --enable-cors
```

But it had no effect and when checking the task definition json under ecs that line has done nothing.


Cheers

**Comment by iamhopaul123** [comment]
> when checking the task definition json under ecs that line has done nothing.

I think for [`command`](https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/#command) it is on the top level. Can you try

```yaml
command: /opt/typesense-server --data-dir /data --api-key=$TYPESENSE_API_KEY --enable-cors
```

**Comment by hailwood** [comment]
Hi @iamhopaul123 we tried command at the top level and it wasn't picked up unfortunately.
